spec-2025x.md.html

```diff
@@ -0,0 +1,1546 @@
+&\qquad\qquad\qquad\quad\ +\,s\, M_{r,ms}(\mathbf{v}, \mathbf{l}; r_u, r_v)\, F_{ms}(F_0 \rho_s, 1)]
+\end{aligned}
+\end{eqnarray}
+
+\begin{equation}
+\begin{aligned}
+&\text{ThinFilmOpaqueDielectric}(\mathbf{v}, \mathbf{l}; r_u, r_v, \rho, \rho_l, s, \eta, \rho_s, l, i, d_i, \eta_i) = \\
+&\qquad\qquad\ (1-i)\, \text{OpaqueDielectric}(\mathbf{v}, \mathbf{l}; r_u, r_v, \rho, \rho_l, s, \eta, \rho_s, l)\\
+&\qquad\quad + \quad\ i\quad\ [B_u(\mathbf{v}, \mathbf{l}; \rho, \rho_l, l)( 1 - \max(R(\mathbf{v}\cdot\mathbf{n};\eta, d_i, \eta_i), R(\mathbf{l}\cdot\mathbf{n};\eta, d_i, \eta_i)))\\
+&\qquad\qquad\qquad\quad\ +\, M_r(\mathbf{v}, \mathbf{l}, \mathbf{h}_r; r_u, r_v)\, R(cos\theta_r, \eta, d_i, \eta_i)\\
+&\qquad\qquad\qquad\quad\ + s \, M_{r,ms}(\mathbf{v}, \mathbf{l}; r_u, r_v)\, F_{ms}(F_0 \rho_s, 1)]
+\end{aligned}
+\end{equation}
+
+With these reformulated components we can also adjust the definitions of the core BSDFs that were defined in Section [Putting It All Together]
+
+\begin{eqnarray}
+&\text{ThinFilmCore}(\rho, \rho_l, s, \eta, \rho_s, i, d_i, \eta_i) &=&\nonumber\\
+&& m & \text{ThinFilmMetallic}(\rho, i, d_i, \eta_i)\\
+&+&(1-m)(1-t) & \text{ThinFilmOpaqueDielectric}(\rho, \rho_l, s, \eta, \rho_s, i, d_i, \eta_i)\nonumber\\
+&+&(1-m)t & \text{ThinFilmTransparentDielectric}(\rho, s, \eta, \rho_s, i, d_i, \eta_i)\nonumber
+\end{eqnarray}
+
+\begin{eqnarray}
+&\text{ThinFilmThinCore}(\rho, s, \rho_s, i, d_i, \eta_i) &=& \nonumber\\
+&&(1-i) & M_r F(\rho_{s,r,0}, \rho_{s,r,90}) \, \nonumber\\
+&+& i (1-t) & M_r\, [(1-m) R(cos\theta_r, \eta, d_i, \eta_i) + m R(cos\theta_r, \eta_\rho, d_i, \eta_i)]\nonumber\\
+&+&& M_{r,ms} F_{ms}(\rho_{s,r,0}, \rho_{s,r,90})\\
+&+&(1-i)&\rho_t {M_t}^\prime(1 - s F(F_0 \rho_s, 1)) \nonumber\\
+&+& i &\rho_t M_t' (1-R(cos\theta_t, \eta, d_i, \eta_i))\nonumber\\
+&+&(1-i)&\rho_d [B_u (1-s) + s B_c(F_0 \rho_s)]\nonumber\\
+&+& i &\rho_d B_u (1 - \max(R(\mathbf{v}\cdot\mathbf{n};\eta, d_i, \eta_i), R(\mathbf{l}\cdot\mathbf{n};\eta, d_i, \eta_i)))\nonumber
+\end{eqnarray}
+
+
+\begin{eqnarray}
+&\text{ThinFilmVolumetricCore}(\rho, s, \eta, \rho_s, i, d_i, \eta_i) &=& \nonumber\\
+&&M_r \, [\quad (1-i)m \, F(\rho, 1)& \nonumber\\
+&&\qquad+\ i\,m\, R(cos\theta_r, \eta_\rho, d_i, \eta_i) \nonumber\\
+&&\qquad+\ (1-i) (1-m) (1-t) s F(F_0 \rho_s, 1) \\
+&&\qquad+\ i\,(1-m) (1-t) R(cos\theta_r, \eta, d_i, \eta_i) \nonumber\\
+&&\qquad+\ (1-m) t s F_d(F_0 \rho_s, 1)] \nonumber\\
+&+& M_{r,ms} \, F_{ms}(\rho_{s,r,0}, \rho_{s,r,90}) \nonumber\\
+&+& (1-i) \rho_t M_t(1 - s F_d(F_0 \rho_s, 1)) \nonumber\\
+&+& i \rho_t M_t(1 - R(cos\theta_t, \eta, d_i, \eta_i)) \nonumber\\
+&+& \rho_d \left[B_u (1-s) + s B_c(F_0 \rho_s)\right]\nonumber \\
+&+& i \rho_d B_u (1 - \max(R(\mathbf{v}\cdot\mathbf{n};\eta, d_i, \eta_i), R(\mathbf{l}\cdot\mathbf{n};\eta, d_i, \eta_i)))\nonumber
+\end{eqnarray}
```

Comment:
In `ThinFilmThinCore` and `ThinFilmVolumetricCore`: Does it look better if you reorder the terms such that all `(1-i)` terms come first, then all `i` terms? Instead of alternating between `(1-i)` and `i`.